### PR TITLE
IMPRO-1809: Independent options for lower and upper single value bins

### PR DESCRIPTION
**Pull request by bayliffe**
Added independent options for including lower and upper single value bins for reliability calibration tables. This enables, for example, the inclusion of only the single value bin corresponding with 0 for use calibrating precipitation.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/calibration/reliability_calibration.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5997286&bid=19663972)

File: improver/cli/construct_reliability_tables.py [hunk 1]
```diff
@@ -64,10 +65,12 @@ def process(
...
-        single_value_limits (bool):
-            Mandates that the extrema bins (0 and 1) should be single valued,
-            with a small precision tolerance of 1.0E-6, e.g. 0 to 1.0E-6 for
-            the lowest bin, and 1 - 1.0E-6 to 1 for the highest bin.
+        single_value_lower_limit (bool):
+            Mandates that the lowest bin should be single valued, with a small
+            precision tolerance, defined as 1.0E-6. The bin is thus 0 to 1.0E-6.
+        single_value_upper_limit (bool):
+            Mandates that the highest bin should be single valued, with a small
+            precision tolerance, defined as 1.0E-6. The bin is thus (1 - 1.0E-6) to 1.
```
Comment (Kat-90) [inline improver/cli/construct_reliability_tables.py:73]: Is this "(1-1.0E-6) to 1" right?

File: improver_tests/calibration/reliability_calibration/test_ConstructReliabilityCalibrationTables.py [hunk 2]
```diff
@@ -173,13 +181,15 @@ def test_without_single_value_limits():
...
+            single_value_lower_limit=False,
+            single_value_upper_limit=False,
         )
         assert_allclose(result, expected)
 
     @staticmethod
-    def test_with_single_value_limits():
-        """Test the generation of probability bins with single value end
+    def test_with_both_single_value_limits():
+        """Test the generation of probability bins with both single value end
```
[the remark this answers is not in the document]
Comment (bayliffe) [reply improver_tests/calibration/reliability_calibration/test_ConstructReliabilityCalibrationTables.py:192]: Done.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/calibration/reliability_calibration.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5997286&bid=19663972)

**Comment by bayliffe** [comment]
@Kat-90 thanks for the suggestion about another acceptance test. It made me notice that I hadn't actually given anyway to turn off the single value bins. This is because in clize, using the ```--single-value-lower-limit``` option for example sets this argument to true. I had also set the default value to True in the CLI, so basically it was always true. To keep everything nicely aligned, i.e. the behaviour of using the CLI with default options and the behaviour of importing and using the plugin with default options, I have modified both the CLI and plugin to have these arguments set to False by default.

I have subsequently tweaked the unit tests, added an extra one for the ```_create_probability_bins_coord``` method, and added the acceptance test you proposed.

@gavinevans you might want to take another look at this as well.

**Pull request merged**